src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1541,1900 @@ interface MPSMatrixMultiplication {
+	{
+		[Export ("pm")]
+		float Pm { get; set; }
+
+		[Export ("ps")]
+		float Ps { get; set; }
+
+		[Export ("p0")]
+		float P0 { get; set; }
+
+		[Export ("kernelWidth")]
+		nuint KernelWidth { get; set; }
+
+		[Export ("kernelHeight")]
+		nuint KernelHeight { get; set; }
+
+		[Static]
+		[Export ("nodeWithSource:kernelSize:")]
+		MPSCnnLocalContrastNormalizationNode Create (MPSNnImageNode sourceNode, nuint kernelSize);
+
+		[Export ("initWithSource:kernelSize:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (MPSNnImageNode sourceNode, nuint kernelSize);
+
+		[Export ("initWithSource:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (MPSNnImageNode sourceNode);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (MPSCnnNormalizationNode), Name = "MPSCNNCrossChannelNormalizationNode")]
+	interface MPSCnnCrossChannelNormalizationNode
+	{
+		[Export ("kernelSizeInFeatureChannels")]
+		nuint KernelSizeInFeatureChannels { get; set; }
+
+		[Static]
+		[Export ("nodeWithSource:kernelSize:")]
+		MPSCnnCrossChannelNormalizationNode Create (MPSNnImageNode sourceNode, nuint kernelSize);
+
+		[Export ("initWithSource:kernelSize:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (MPSNnImageNode sourceNode, nuint kernelSize);
+
+		[Export ("initWithSource:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (MPSNnImageNode sourceNode);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSNnFilterNode), Name = "MPSNNScaleNode")]
+	interface MPSNnScaleNode
+	{
+		[Static]
+		[Export ("nodeWithSource:outputSize:")]
+		MPSNnScaleNode Create (MPSNnImageNode sourceNode, MTLSize size);
+
+		[Static]
+		[Export ("nodeWithSource:transformProvider:outputSize:")]
+		MPSNnScaleNode Create (MPSNnImageNode sourceNode, [NullAllowed] IMPSImageTransformProvider transformProvider, MTLSize size);
+
+		[Export ("initWithSource:outputSize:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode, MTLSize size);
+
+		[Export ("initWithSource:transformProvider:outputSize:")]
+		IntPtr Constructor (MPSNnImageNode sourceNode, [NullAllowed] IMPSImageTransformProvider transformProvider, MTLSize size);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSNnFilterNode), Name = "MPSNNBinaryArithmeticNode")]
+	interface MPSNnBinaryArithmeticNode
+	{
+		[Static]
+		[Export ("nodeWithSources:")]
+		MPSNnBinaryArithmeticNode Create (MPSNnImageNode[] sourceNodes);
+
+		[Static]
+		[Export ("nodeWithLeftSource:rightSource:")]
+		MPSNnBinaryArithmeticNode Create (MPSNnImageNode left, MPSNnImageNode right);
+
+		[Export ("initWithSources:")]
+		IntPtr Constructor (MPSNnImageNode[] sourceNodes);
+
+		[Export ("initWithLeftSource:rightSource:")]
+		IntPtr Constructor (MPSNnImageNode left, MPSNnImageNode right);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSNnFilterNode), Name = "MPSNNConcatenationNode")]
+	interface MPSNnConcatenationNode
```

Comment:
unless there are subclasses, this looks immutable so I would guess init (default constructor) does not do any good -> [DisableDefaultCtor]

Also `Nn` -> `NN`